cmd/object-handlers-common.go

```diff
@@ -319,11 +326,23 @@ func checkPreconditions(ctx context.Context, w http.ResponseWriter, r *http.Requ
 	return false
 }
 
+// checkIfUnmodifiedSince - return true if object was modified after(Including critical values) ifUnmodifiedSinceHeader time
+// return false if the header of If-Unmodified-Since was empty or object was modified before(Excluding critical values) ifUnmodifiedSinceHeader time
+func checkIfUnmodifiedSince(r *http.Request, objInfo ObjectInfo) bool {
+	ifUnmodifiedSinceHeader := r.Header.Get(xhttp.IfUnmodifiedSince)
+	if ifUnmodifiedSinceHeader != "" {
+		if givenTime, err := amztime.ParseHeader(ifUnmodifiedSinceHeader); err == nil {
+			return !ifModifiedSince(objInfo.ModTime, givenTime)
+		}
+	}
+	return false
+}
+
 // returns true if object was modified after givenTime.
 func ifModifiedSince(objTime time.Time, givenTime time.Time) bool {
-	// The Date-Modified header truncates sub-second precision, so
-	// use mtime < t+1s instead of mtime <= t to check for unmodified.
-	return objTime.After(givenTime.Add(1 * time.Second))
+	// return true if givenTime < objTime (object has been modified since givenTime)
+	// return false if givenTime >= objTime (object has not been modified since givenTime)
+	return givenTime.Before(objTime)
```

Comment:
The comments above this return are not necessary as they are just describing exactly what the code does, they don't add any useful information or explain any context. Better to just remove them.

Reply:
okay.